services/queue/src/artifacts.js

```diff
@@ -216,51 +226,48 @@ builder.declare({
       throw err;
     }
 
-    // Load existing Artifact entity
-    artifact = artifactUtils.fromDbRows(await this.db.fns.get_queue_artifact(taskId, runId, name));
+    // Load original Artifact entity
+    const original = artifactUtils.fromDbRows(await this.db.fns.get_queue_artifact(taskId, runId, name));
 
-    // Allow recreating of the same artifact, report conflict if it's not the
-    // same artifact (allow for later expiration).
-    // Note, we'll check `details` later
-    if (artifact.storageType !== storageType ||
-      artifact.contentType !== contentType ||
-      artifact.expires.getTime() > expires.getTime()) {
-      return res.reportError('RequestConflict',
-        'Artifact already exists, with different type or later expiration\n\n' +
-        'Existing artifact information: {{originalArtifact}}', {
-          originalArtifact: {
-            storageType: artifact.storageType,
-            contentType: artifact.contentType,
-            expires: artifact.expires,
-          },
-        });
-    }
+    let ok = true;
+    ok = ok && original.storageType === storageType;
```

Comment:
Interesting way to break up a long conditional, very readable. I miss `&=` and `|=`.